pyproject.toml

```diff
@@ -151,11 +151,11 @@ pytest-clarity = "^1.0.1"
 optional = true
 
 [tool.poetry.group.dev.dependencies]
-ruff = "^0.0.286"
-black = "^23.7.0"
-pyright = "^1.1.355"
-pre-commit = "^3.5.0"
-boto3-stubs = {extras = ["bedrock", "iam", "opensearch", "s3", "sagemaker"], version = "^1.34.18"}
+ruff = "^0.4.4"
+black = "^23.12.1"
```

Comment:
Do we still need black?

[Ruff](https://docs.astral.sh/ruff/) claims:
> Drop-in parity with [Flake8](https://docs.astral.sh/ruff/faq/#how-does-ruff-compare-to-flake8), isort, and Black

Reply:
Oh dang that's new(ish)! Goodbye black 🫡